pkg/install/deploystorage.go

```diff
@@ -256,47 +256,38 @@ func (i *Installer) deployStorageTemplate(ctx context.Context, installConfig *in
 		return err
 	}
 
-	var g graph
-
 	exists, err := i.graphExists(ctx)
-	if err != nil {
+	if err != nil || exists {
 		return err
 	}
 
-	if exists {
-		g, err = i.loadGraph(ctx)
-		if err != nil {
-			return err
-		}
-
-	} else {
-		clusterID := &installconfig.ClusterID{
-			UUID:    i.doc.ID,
-			InfraID: infraID,
-		}
-
-		g = graph{
-			reflect.TypeOf(installConfig): installConfig,
-			reflect.TypeOf(platformCreds): platformCreds,
-			reflect.TypeOf(image):         image,
-			reflect.TypeOf(clusterID):     clusterID,
-		}
+	clusterID := &installconfig.ClusterID{
+		UUID:    i.doc.ID,
+		InfraID: infraID,
+	}
 
-		i.log.Print("resolving graph")
-		for _, a := range targets.Cluster {
-			_, err := g.resolve(a)
-			if err != nil {
-				return err
-			}
-		}
+	g := graph{
+		reflect.TypeOf(installConfig): installConfig,
+		reflect.TypeOf(platformCreds): platformCreds,
+		reflect.TypeOf(image):         image,
+		reflect.TypeOf(clusterID):     clusterID,
+	}
 
-		// the graph is quite big so we store it in a storage account instead of in cosmosdb
-		err = i.saveGraph(ctx, g)
+	i.log.Print("resolving graph")
+	for _, a := range targets.Cluster {
+		_, err := g.resolve(a)
 		if err != nil {
 			return err
 		}
 	}
 
+	// the graph is quite big so we store it in a storage account instead of in cosmosdb
+	return i.saveGraph(ctx, g)
+}
+
+func (i *Installer) attachNSGsAndPatch(ctx context.Context) error {
```

Comment:
As part of moving as many timeouts and authorization error checks as possible _out_ of the steps themselves, `deployStorageTemplate` was split up here.